bin/deploy-vhost

```diff
@@ -323,6 +348,26 @@ END
 END
             }
     }
+
+    # Add all the parameters as structured data
+
+    push $conf->{struct_database_configs}{internal}, (
```

Comment:
So `$conf->{struct_database_configs}{internal}` is a scalar (which you want to be a reference to an array), @{$conf->{struct_database_configs}{internal}}` is the array that reference points to. And then you need to push a hashref `{…}` rather than an array `(…)` because pushing onto an array just appends the items to the end, and you want an array of independent database config hashes, not one merged list of the configs.